ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/sync/worldstate/WorldDownloadStateTest.java

```diff
@@ -131,6 +136,18 @@ public void shouldResetRequestsSinceProgressCountWhenProgressIsMade() {
     assertThat(downloadState.getDownloadFuture()).isCompletedExceptionally();
   }
 
+  @Test
+  public void shouldNotBeStalledWhenMaxRequestsReachedButNotMinimumTime() {
```

Comment:
```suggestion
  public void shouldStallWhenMaxRequestsReachedAndTimeLimitExceeded() {
```

Reply:
Looks like this test is checking for stalling?  Worth adding the tests for the 2 other cases where it doesn't stall though (T/F and F/T for timeLimitHit/failedRequestsLimitHit).

Reply:
The test hits the maximum requests without progress limit and checks the download isn't considered stalled until the minimum time condition is also met.  Tidied up the test name and implementation to make this clearer.

Reply:
Ensured the other combinations of time/requests reached/not-reached were covered.